test/test_c10d.py

```diff
@@ -3205,6 +3206,31 @@ def test_broadcast_coalesced_gloo_cpu(self):
         device = torch.device('cpu')
         self._test_broadcast_coalesced(process_group, device)
 
+class BackendLoadingTest(TestCase):
+    def _load_c10d_test_backend(self):
+        src = "{}/cpp_extensions/cpp_c10d_extension.cpp".format(os.path.abspath(os.path.dirname(__file__)))
+        extension = load(
+            name="c10dtest",
+            sources=[src],
+        )
+
+    def test_collective_api(self):
+        self._load_c10d_test_backend()
+
+        os.environ['WORLD_SIZE'] = '1'
+        os.environ['MASTER_ADDR'] = '127.0.0.1'
+        os.environ['MASTER_PORT'] = str(common.find_free_port())
+        os.environ['RANK'] = '0'
+
+        c10d.init_process_group(backend='c10dtest', init_method='env://', world_size=1, rank=0)
+        self.assertEqual(c10d.get_rank(), 0)
+        self.assertEqual(c10d.get_world_size(), 1)
+
+        with tempfile.NamedTemporaryFile(delete=False) as file:
+            store = c10d.FileStore(file.name, 1)
+            process_group = c10d.distributed_c10d._get_default_group()
+            work = process_group.allreduce(torch.rand(1))
```

Comment:
do we need to add some assert for this block of test?

Reply:
done by adding a dumy work and wait work success